sessionctx/variable/sysvar.go

```diff
@@ -135,7 +135,13 @@ type SysVar struct {
 func (sv *SysVar) GetGlobalFromHook(s *SessionVars) (string, error) {
 	// Call the Getter if there is one defined.
 	if sv.GetGlobal != nil {
-		return sv.GetGlobal(s)
+		val, err := sv.GetGlobal(s)
+		if err != nil {
+			return val, err
+		}
+		// Ensure that the results from the getter are validated
+		// Since some are read directly from tables.
+		return sv.ValidateWithRelaxedValidation(s, val, ScopeGlobal), nil
```

Comment:
This is required because in many case getters are reading from table values and accept user input directly without normalization. `ValidateWithRelaxedValidation` is the same function that is called when init'ing a new session and assigning values to the session variables.